packages/atomic/src/components/facets/atomic-category-facet/atomic-category-facet.tsx

```diff
@@ -156,6 +160,15 @@ export class AtomicCategoryFacet
   @Prop() public injectionDepth = 1000;
   // @Prop() public customSort?: string; TODO: KIT-753 Add customSort option for facet
 
+  @FocusTarget()
+  protected showMoreFocus!: FocusTargetController;
```

Comment:
This is a good example of what `@FocusTarget` does.

When we press "show less", we know that we want to focus on the "show more" button after. However, if somebody pressed "show more" too many times and all children are shown, then the "show more" button doesn't exist until we press on "show less". Additionally, buttons will only be removed and added after a new search was rendered.

The decorator solves that issue by diving the whole process into two methods:
* setTarget
* focusAfterSearch

Calling `focusAfterSearch` activates a flag that indicates that we want to focus on the target as soon as:
* It's available
* A new search was performed

Calling `setTarget` will set the target to be focused. If the conditions for `focusAfterSearch` are met, it will also focus on the target immediately.